core/model/modx/processors/search/search.class.php

```diff
@@ -18,37 +18,651 @@ class modSearchProcessor extends modProcessor
+            $like = ($search != $query[3]) ? ':LIKE' : ':LIKE';            
+
+            foreach($fields as $fieldname){
+                $n++;
+                if($class_key == 'modUser' AND $fieldname != 'username'){
+                    if($n == 1)
+                    {
+                        $key = 'Profile.'.$fieldname.$like;                     
+                    }
+                    else
+                    {
+                        $key = 'OR:.Profile'.$fieldname.$like;
+                    }
+                }
+                else {
+                    if($n == 1)
+                    {
+                        $key = $fieldname.$like;
+                        
+                    }
+                    else
+                    {
+                        $key = 'OR:'.$fieldname.$like;
+                    }
+                }
+                $where[$key] = $search;
+            }
+            $c->where($where);
+        }            
+        
+        if($debug === true)
+        {
+            $debug['q'] = '<br/>Full query: <pre style="color: #fff">' . print_r($where,true) . '</pre>';
+        }
+        
+        $totalResults = count($this->modx->getCollection($class_key, $c));
+        
+        $offset  = is_numeric($query[4]) && $query[4] > 0 ? ( $query[4] - 1 ) * $maxResults : 0;
+        $start = ($totalResults) ? $offset + 1 : 0;
+        
+        if($offset == 0){
+            $end = $maxResults > $totalResults ? $totalResults: $maxResults;    
+        }
+        else {
+            $end = ($offset * $query[4] < $totalResults) ? $offset * $query[4] : $totalResults;
+        }
+        
+            
+        $pages = ceil($totalResults / $maxResults) ;
+        
+        switch(count($query)){
+            case 4:
+                $header = $this->modx->lexicon('uberbar_enter_searchstring_header');
+                break;
+            default:
+                if($totalResults > $maxResults)
+                {
+                    $activePage = !empty($query[4]) ? $query[4] : '1';
+                    for($page = 1; $page <= $pages; $page++){
+                        $activeClass = ($page == $activePage ) ? 'active ' : '';
+                        $options .= '<li><a data-value="' . $page . '" class="' .$activeClass . $btn . '">' . $page . '</a></li>';
+                    }
+                }                
+                $header = $this->modx->lexicon('uberbar_searchresults_header');
+                break;
+        }
+        
+        $exec = number_format(microtime(true) - $startTime , 3);
+        
+        if($totalResults > 0)
+        {
+            //'Your search returned <strong>' . $totalResults . ' results </strong> in ' . $exec . ' seconds.';
+            
+            $s = array('{totalResults}','{exec}');
+            $r = array($totalResults,$exec);
+            $msg = str_replace( $s , $r , $this->modx->lexicon('uberbar_searchresults_yes' ) );
```

Comment:
As with the earlier comment, this can also simply pass an associative array into the lexicon method to use placeholders.